model/model.go

```diff
@@ -181,6 +181,22 @@ func (t *TableInfo) GetPkColInfo() *ColumnInfo {
 	return nil
 }
 
+// Cols returns the columns of the table in public state.
+func (t *TableInfo) Cols() []*ColumnInfo {
```

Comment:
This will be called many times. Could you pre-calculate this?

Reply:
you can save the publicColumns in TableInfo.

Reply:
It's only called in `ToPB`.
Later we can change the order of columns in DDL, then we don't need to allocate memories.